src/test/java/de/medizininformatikinitiative/torch/BundleIT.java

```diff
@@ -0,0 +1,275 @@
+
+        return mapper.writeValueAsString(node);
+    }
+
+
+    private static Bundle requestCoreBundle(String url) {
+        var response = nginxClient.get().uri(url).retrieve().toEntity(String.class).block();
+        return (Bundle) parser.parseResource(response.getBody());
+    }
+
+    private static List<Bundle> requestPatientBundles(List<String> urls) {
+        List<Bundle> bundles = new LinkedList<>();
+        for (String url : urls) {
+            var response = nginxClient.get().uri(url).retrieve().toEntity(String.class).block();
+            var scanner = new Scanner(response.getBody());
+            while (scanner.hasNextLine()) {
+                bundles.add((Bundle) parser.parseResource(scanner.nextLine()));
+            }
+        }
+        return bundles;
+    }
+
+    private static List<String> retrieveBundles(String crtdlFile) throws IOException {
+        var crtdl = loadCrtdl(crtdlFile);
+        var statusResponse = torchClient.post().bodyValue(crtdl).retrieve().toEntity(String.class).block();
+        var bundleLocationResponse = nginxClient.get().uri(getStatusUrl(statusResponse)).retrieve().toEntity(String.class).block();
+        return getOutputUrls(bundleLocationResponse);
+    }
+
+    @Test
+    public void testExamples() throws IOException {
+
+        var bundleUrls = retrieveBundles("it-kds-crtdl.json");
+        var coreBundle = requestCoreBundle(bundleUrls.getFirst());
+        var patientBundles = requestPatientBundles(bundleUrls.subList(1, bundleUrls.size()));
+
+        assertThat(coreBundle).containsNEntries(0);
+        assertThat(patientBundles).hasSize(2);
+
+        // test if referenced IDs match the actual patient's IDs
+        assertThat(patientBundles).allSatisfy(bundle -> assertThat(bundle).extractOnlyPatient().satisfies(patient ->
+                assertThat(bundle).extractResources(r -> r.getResourceType().equals(ResourceType.Condition)).allSatisfy(condition ->
+                        assertThat(condition).extractChildrenStringsAt("subject.reference").hasSize(1).first().isEqualTo(patient.getId()))));
+
+        // this is equivalent to 'never exists' (and an example how to filter by resource type)
+        assertThat(patientBundles).allSatisfy(b -> assertThat(b).extractResources(r -> r.getResourceType().equals(ResourceType.Condition))
+                .allSatisfy(c -> assertThat(c).extractElementsAt("bodySite").isEmpty()));
+
+        // opposite of 'never exists'
+        assertThat(patientBundles).allSatisfy(b -> assertThat(b).extractResources(r -> r.getResourceType().equals(ResourceType.Condition))
+                .allSatisfy(
+                        r -> assertThat(r).extractElementsAt("code").isNotEmpty()
+                ));
+
+        // all conditions in all patient bundles must have the given top fields
+        assertThat(patientBundles).allSatisfy(b -> assertThat(b).extractResources(r -> r.getResourceType().equals(ResourceType.Condition))
```

Comment:
@bastianschaffer , @alexanderkiel  - i think it might be a good idea to encapsulate the type into the function as we will moslty need that
=> extractResourcesByType() - and potentially if no type is given = all types